test/UnitTest/Microsoft.AspNetCore.OData.Test/Routing/ODataActionSelectorTest.cs

```diff
@@ -0,0 +1,249 @@
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNet.OData.Routing;
+using Microsoft.AspNet.OData.Test.Abstraction;
+using Microsoft.AspNet.OData.Query;
+using Microsoft.AspNet.OData.Test.Common;
+using Xunit;
+using System.Threading;
+
+namespace Microsoft.AspNet.OData.Test.Routing
+{
+    public class ODataActionSelectorTest
+    {
+
+        //public static TheoryDataSet<Dictionary<string, object>, string, string, Type, string, Type[]> ScenariosWithCorrectAction
+        public static TheoryDataSet<Dictionary<string, object>, (string Method, string Body), (Type ControllerType, string ActionName, Type[] Signature)> ScenariosWithCorrectAction
```

Comment:
The tuple was meant to organize related parameters together, but also to avoid having to many type parameters to the `TheoryDataSet` generic class because it only supports a limited number of type parameters.